### ISSUE TITLE
Evaluating model using model.evaluate(data) returns empty list (using Keras MNIST VAE example)

### ISSUE THREAD

**Issue by preethiseshadri518** [keras-team/keras-io#232]
Hi @fchollet! First off, thanks for the example code. It has been super helpful for a beginner like me! 

I am following the VAE tutorial found here: https://keras.io/examples/generative/vae/, which defines a custom VAE class. I followed the tutorial completely, and also added a `test_step()` method so that I could evaluate validation data performance during training, as shown in the discussion here: https://github.com/keras-team/keras-io/issues/38. 

I now want to evaluate the model on various datasets after training and plot the corresponding losses. When I try evaluating the model using `vae.evaluate(data)`, where data is a subset of the test dataset,  I get an empty list `[]`. Since the `evaluate()` function uses `verbose=True` by default, it still prints the following: `33/33 [==============================] - 0s 3ms/step - loss: 46.9414 - reconstruction_loss: 44.1429 - kl_loss: 2.7984`. 

How do I fix this? It seems like evaluate() is capable of computing the various loss metrics but for some reason is just not returning them. Do I need to implement a new evaluate() method because this is a custom model? Note: I know I can use the `test_step()` function to compute the loss metrics, which is my current work around, but I just want to understand why evaluate() does not work. 

Here is the VAE class code (which comes from the tutorial linked above):

```
`class VAE(keras.Model):
    def __init__(self, encoder, decoder, **kwargs):
        super(VAE, self).__init__(**kwargs)
        self.encoder = encoder
        self.decoder = decoder


    def train_step(self, data):
        if isinstance(data, tuple):
            data = data[0]
        with tf.GradientTape() as tape:
            z_mean, z_log_var, z = self.encoder(data)
            reconstruction = self.decoder(z)
            reconstruction_loss = tf.reduce_mean(
                keras.losses.mean_squared_error(data, reconstruction)
            )
            reconstruction_loss *= 28 * 28
           
            kl_loss = 1 + z_log_var - tf.square(z_mean) - tf.exp(z_log_var) 
            kl_loss = tf.reduce_mean(kl_loss)
            kl_loss *= -0.5 

            total_loss = reconstruction_loss + kl_loss 
        grads = tape.gradient(total_loss, self.trainable_weights)
        self.optimizer.apply_gradients(zip(grads, self.trainable_weights))

        return {
            "loss": total_loss,
            "reconstruction_loss": reconstruction_loss,
            "kl_loss": kl_loss,
        }


    def test_step(self, data):
      if isinstance(data, tuple):
        data = data[0]

      z_mean, z_log_var, z = self.encoder(data)
      reconstruction = self.decoder(z)
      reconstruction_loss = tf.reduce_mean(
            keras.losses.mean_squared_error(data, reconstruction)
      )
      reconstruction_loss *= 28 * 28

      kl_loss = 1 + z_log_var - tf.square(z_mean) - tf.exp(z_log_var)
      kl_loss = tf.reduce_mean(kl_loss)
      kl_loss *= -0.5
      total_loss = reconstruction_loss + kl_loss

      return {
          "loss": total_loss,
          "reconstruction_loss": reconstruction_loss,
          "kl_loss": kl_loss,
      }`
```

**Comment by fchollet** [comment]
This is a bug, we're fixing it. As a workaround for now, you can call `evaluate(data, return_dict=True)` which will give you the full dict of metrics.

**Comment by fchollet** [comment]
This will be fixed in the *next* tf-nightly build.

**Comment by diman82** [comment]
Has it been fixed for TF 2.4.0? I'm facing the same issue, and the suggested workaround didn't help..
